tlscommon.c

```diff
@@ -626,13 +626,41 @@ static int certattr_matchip(GENERAL_NAME *gn, struct certattrmatch *match){
         && !memcmp(ASN1_STRING_get0_data(gn->d.iPAddress), &match->ipaddr, l)) ? 1 : 0 ;
 }
 
+// Generates a generic DNS. eg: converts idp.openroaming.net to *.openroaming.net
+// Compares generated generic DNS with DNS from certificate
+// returns 1 if both match else returns 0
+static int compareWithModifiedHostname(char* certDNS, char* scriptHostName) {
+    int len = 0;
+    int lenOfGeneric = 0, lenOfModifiedScriptHostName = 0;
+    char* pos;
+    char* modifiedScriptHostName;
+    char *result;
+    char generic[] = "*";
+    pos = strchr(scriptHostName, '.');
```

Comment:
This func is called from _general_name_regex_match so it might be possible to pass an IPv6 address in, which will not contain a '.'  If it is not found, then immediately exit with 0.